src/build.rs

```diff
@@ -295,6 +312,70 @@ impl<'cb> CheckoutBuilder<'cb> {
         self.flag(raw::GIT_CHECKOUT_CONFLICT_STYLE_MERGE, on)
     }
 
+    /// Indicate whether a notification callback should never be invoked.
+    ///
+    /// This is the default.
+    pub fn never_notify(&mut self) -> &mut CheckoutBuilder<'cb> {
+        self.notify_flags = CheckoutNotificationType::empty();
+        self
+    }
+
+    /// Invoke a notification callback for every notification type.
+    pub fn always_notify(&mut self) -> &mut CheckoutBuilder<'cb> {
+        self.notify_flags = CheckoutNotificationType::all();
+        self
+    }
+
+    fn notify_flag(&mut self, bit: CheckoutNotificationType,
+            on: bool) -> &mut CheckoutBuilder<'cb> {
+        if on {
+            self.notify_flags.insert(bit);
+        } else {
+            self.notify_flags.remove(bit);
+        }
+        self
+    }
+
+    /// Indicate whether a notification callback should be invoked for
+    /// notifications of conflicts.
+    ///
+    /// Defaults to false.
+    pub fn notify_on_conflict(&mut self, on_conflict: bool) -> &mut CheckoutBuilder<'cb> {
+        self.notify_flag(CHECKOUT_NOTIFICATION_CONFLICT, on_conflict)
+    }
+
+    /// Indicate whether a notification callback should be invoked for
+    /// notifications of dirty files.
+    ///
+    /// Defaults to false.
+    pub fn notify_on_dirty(&mut self, on_dirty: bool) -> &mut CheckoutBuilder<'cb> {
+        self.notify_flag(CHECKOUT_NOTIFICATION_DIRTY, on_dirty)
+    }
+
+    /// Indicate whether a notification callback should be invoked for
+    /// notifications of changed files.
+    ///
+    /// Defaults to false.
+    pub fn notify_on_updated(&mut self, on_updated: bool) -> &mut CheckoutBuilder<'cb> {
+        self.notify_flag(CHECKOUT_NOTIFICATION_UPDATED, on_updated)
+    }
+
+    /// Indicate whether a notification callback should be invoked for
+    /// notifications of untracked files.
+    ///
+    /// Defaults to false.
+    pub fn notify_on_untracked(&mut self, on_untracked: bool) -> &mut CheckoutBuilder<'cb> {
+        self.notify_flag(CHECKOUT_NOTIFICATION_UNTRACKED, on_untracked)
+    }
+
+    /// Indicate whether a notification callback should be invoked for
+    /// notifications of ignored files.
+    ///
+    /// Defaults to false.
+    pub fn notify_on_ignored(&mut self, on_ignored: bool) -> &mut CheckoutBuilder<'cb> {
+        self.notify_flag(CHECKOUT_NOTIFICATION_IGNORED, on_ignored)
+    }
```

Comment:
Yeah! That was what I was going to do, as I mentioned in my post, but I thought perhaps since there were other functions for specific flags that you wanted to do the same here. I'm going to create one called `.notify_on(t: CheckoutNotificationType)`.